scenarios/telemetry/dotnet/telemetry_consumer/Worker.cs

```diff
@@ -14,36 +13,44 @@ public class Worker : BackgroundService
     public Worker(ILogger<Worker> logger, IConfiguration configuration)
     {
         _logger = logger;
-        _configuration  = configuration;
+        _configuration = configuration;
     }
 
     protected override async Task ExecuteAsync(CancellationToken stoppingToken)
     {
         var cs = MqttConnectionSettings.CreateFromEnvVars(_configuration.GetValue<string>("envFile")!);
         _logger.LogInformation("Connecting to {cs}", cs);
 
-        var mqttClient = new MqttFactory().CreateManagedMqttClient(MqttNetTraceLogger.CreateTraceLogger());
+        var mqttClient = new MqttFactory().CreateMqttClient(MqttNetTraceLogger.CreateTraceLogger());
 
-        mqttClient.InternalClient.ConnectedAsync += async cea =>
+        mqttClient.DisconnectedAsync += async d =>
         {
-            _logger.LogInformation("Client {ClientId} connected: {ResultCode}", mqttClient.InternalClient.Options.ClientId, cea.ConnectResult.ResultCode);
+            await Task.Yield();
+            _logger.LogWarning("Disconnected from server with reason {reason} {ex}", d.Reason, d.Exception?.Message);
+        };
+
+        var connAck = await mqttClient.ConnectAsync(new MqttClientOptionsBuilder()
+            .WithConnectionSettings(cs)
+            .Build(), stoppingToken);
+
+        _logger.LogInformation("Client {ClientId} connected: {ResultCode}, session {s}", mqttClient.Options.ClientId, connAck.ResultCode, connAck.IsSessionPresent);
 
-            PositionTelemetryConsumer positionTelemetry = new(mqttClient.InternalClient)
+        PositionTelemetryConsumer positionTelemetry = new(mqttClient)
+        {
+            OnTelemetryReceived = async m =>
             {
-                OnTelemetryReceived = m => 
-                _logger.LogInformation("Received msg from {id}. Coordinates lat: {x}, lon: {y}",
-                        m.ClientIdFromTopic, 
-                        m.Payload!.Coordinates.Latitude, 
-                        m.Payload.Coordinates.Longitude)
-            };
-            await positionTelemetry.StartAsync();
+                await Task.Yield();
+
+                _logger.LogInformation("Received msg '{mid}' from {client}. Coordinates lat: {x}, lon: {y}",
+                    m.MessageId,
+                    m.ClientIdFromTopic,
+                    m.Payload!.Coordinates.Latitude,
+                    m.Payload.Coordinates.Longitude);
```

Comment:
Does this reference need to include the ~ and be `m.Payload!.Coordinates.Longitude` as well?

Reply:
yes